README.md
# NAME

Filter::CommaEquals - Adds support for ,= to any package or script

# VERSION

version 1.02

[![test](https://github.com/gryphonshafer/Filter-CommaEquals/workflows/test/badge.svg)](https://github.com/gryphonshafer/Filter-CommaEquals/actions?query=workflow%3Atest)
[![codecov](https://codecov.io/gh/gryphonshafer/Filter-CommaEquals/graph/badge.svg)](https://codecov.io/gh/gryphonshafer/Filter-CommaEquals)

# SYNOPSIS

    use Filter::CommaEquals;
    my @array = ( 42, 1138, 96 );

    @array ,= 433;
    # exactly the same as writing: push( @array, 433 );

    print join(', ', @array), "\n";
    # prints: 42, 1138, 96, 433

# DESCRIPTION

Adds support for ,= to any package or script. Perl has +=, -=, /=, \*=, .=, and
so forth to operate on scalars, but it doesn't have ,= to operate on arrays.
This module effectively lets you rewrite push statements with ,= instead.

For example...

    push @array, $element;
    push( @array, $element_1, $element_2, $element_3 );
    push @array, [ 42, 1138, 96, 433 ];

...can now be rewritten as...

    use Filter::CommaEquals;
    @array ,= $element;
    @array ,= $element_1, $element_2, $element_3;
    @array ,= [ 42, 1138, 96, 433 ];

Cool, huh? Admit it. You want to write ,= instead of push, don't you.
You can save typing 3 whole characters!

Filter::CommaEquals is scoped to the package or script that it's used in,
but nothing more, and it requires Perl version 5.7.1 or higher.

# MOTIVATION

A coworker complained about ,= not being in core Perl. After some thought,
I realized writing ,= is faster (by 3 key presses) than push. Also, I'm lazy.

# SEE ALSO

You can also look for additional information at:

- [GitHub](https://github.com/gryphonshafer/Filter-CommaEquals)
- [MetaCPAN](https://metacpan.org/pod/Filter::CommaEquals)
- [GitHub Actions](https://github.com/gryphonshafer/Filter-CommaEquals/actions)
- [Codecov](https://codecov.io/gh/gryphonshafer/Filter-CommaEquals)
- [CPANTS](http://cpants.cpanauthors.org/dist/Filter-CommaEquals)
- [CPAN Testers](http://www.cpantesters.org/distro/F/Filter-CommaEquals.html)

# AUTHOR

Gryphon Shafer <gryphon@cpan.org>

# COPYRIGHT AND LICENSE

This software is Copyright (c) 2014-2021 by Gryphon Shafer.

This is free software, licensed under:

    The Artistic License 2.0 (GPL Compatible)


Changes
Revision history for Filter::CommaEquals

{{$NEXT}}

1.02      2021-01-10 14:09:56-08:00 America/Los_Angeles
        - Migrate to Test2::V0
        - Improve .gitignore file
        - License migrate to Artistic 2.0
        - Migrate Travis CI and Coveralls to GitHub Actions and Codecov
        - Update copyright year
        - Update Travis YAML
        - Adding Perl versions to test; annual general update for 2018

1.01      2015-01-29 14:39:41-08:00 America/Los_Angeles
        - Implementing use of Dist::Zilla

0.03      Wed Sep 25 10:00:58 PDT 2014
        - Some OCD cleanup; addition of Travis and Coveralls

0.02      Sun Aug 10 00:12:00 PDT 201
        - Code style updating

0.01      Thu Jul 14 15:45:31 2005
        - Original version


lib/Filter/CommaEquals.pm
package Filter::CommaEquals;
# ABSTRACT: Adds support for ,= to any package or script

use 5.008;
use strict;
use warnings;

use Filter::Simple;

# VERSION

FILTER_ONLY
    code => sub {
        s/(\@[^',=']+)\s*,=\s*([^;]+);/push $1, $2;/msg
    };

1;
__END__

=pod

=begin :badges

=for markdown
[![test](https://github.com/gryphonshafer/Filter-CommaEquals/workflows/test/badge.svg)](https://github.com/gryphonshafer/Filter-CommaEquals/actions?query=workflow%3Atest)
[![codecov](https://codecov.io/gh/gryphonshafer/Filter-CommaEquals/graph/badge.svg)](https://codecov.io/gh/gryphonshafer/Filter-CommaEquals)

=end :badges

=begin :prelude

=for test_synopsis BEGIN { die "SKIP: skip Test::Synopsis because content is post-filter\n"; }

=end :prelude

=head1 SYNOPSIS

    use Filter::CommaEquals;
    my @array = ( 42, 1138, 96 );

    @array ,= 433;
    # exactly the same as writing: push( @array, 433 );

    print join(', ', @array), "\n";
    # prints: 42, 1138, 96, 433

=head1 DESCRIPTION

Adds support for ,= to any package or script. Perl has +=, -=, /=, *=, .=, and
so forth to operate on scalars, but it doesn't have ,= to operate on arrays.
This module effectively lets you rewrite push statements with ,= instead.

For example...

    push @array, $element;
    push( @array, $element_1, $element_2, $element_3 );
    push @array, [ 42, 1138, 96, 433 ];

...can now be rewritten as...

    use Filter::CommaEquals;
    @array ,= $element;
    @array ,= $element_1, $element_2, $element_3;
    @array ,= [ 42, 1138, 96, 433 ];

Cool, huh? Admit it. You want to write ,= instead of push, don't you.
You can save typing 3 whole characters!

Filter::CommaEquals is scoped to the package or script that it's used in,
but nothing more, and it requires Perl version 5.7.1 or higher.

=head1 MOTIVATION

A coworker complained about ,= not being in core Perl. After some thought,
I realized writing ,= is faster (by 3 key presses) than push. Also, I'm lazy.

=head1 SEE ALSO

You can also look for additional information at:

=for :list
* L<GitHub|https://github.com/gryphonshafer/Filter-CommaEquals>
* L<MetaCPAN|https://metacpan.org/pod/Filter::CommaEquals>
* L<GitHub Actions|https://github.com/gryphonshafer/Filter-CommaEquals/actions>
* L<Codecov|https://codecov.io/gh/gryphonshafer/Filter-CommaEquals>
* L<CPANTS|http://cpants.cpanauthors.org/dist/Filter-CommaEquals>
* L<CPAN Testers|http://www.cpantesters.org/distro/F/Filter-CommaEquals.html>

=cut


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.017.
use strict;
use warnings;

use 5.008;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Adds support for ,= to any package or script",
  "AUTHOR" => "Gryphon Shafer <gryphon\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Filter-CommaEquals",
  "LICENSE" => "artistic_2",
  "MIN_PERL_VERSION" => "5.008",
  "NAME" => "Filter::CommaEquals",
  "PREREQ_PM" => {
    "Filter::Simple" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => 0,
    "Test::Most" => 0
  },
  "VERSION" => "1.02",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "File::Spec" => 0,
  "Filter::Simple" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Test::More" => 0,
  "Test::Most" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
